Modules/_decimal/_decimal.c

```diff
@@ -1754,6 +1782,17 @@ ctxmanager_new(PyTypeObject *type UNUSED, PyObject *args, PyObject *kwds)
     self->global = global;
     Py_INCREF(self->global);
 
+    int ret = context_setattrs(
+        self->local, prec, rounding,
+        Emin, Emax, capitals,
+        clamp, flags, traps
+    );
+
+    if(ret < 0){
+        return NULL;
+    };
```

Comment:
```suggestion
    if (ret < 0) {
        return NULL;
    }
```

Also, this needs to clean up self, right now it's being leaked.